### PR TITLE
*: cut 2.7.0-rc.0

### PR DESCRIPTION
**Pull request by gouthamve**
Little late, I realise, sorry about that!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+* [FEATURE] Add CORS origin flag. #5011
+* [ENHANCEMENT] TSDB: Run CI tests on Windows too to catch windows specific errors. prometheus/tsdb#469
+* [ENHANCEMENT] Consul SD: Add tagged address to the discovery metadata. #5001
+* [ENHANCEMENT] Kubernetes SD: Add service external IP and external name to the discovery metadata. #4940
+* [ENHANCEMENT] Azure SD: Add support for Managed Identity authentication. #4590
+* [ENHANCEMENT] Azure SD: Add tenant and subscription IDs to the discovery metadata. #4969
+* [ENHANCEMENT] OpenStack SD: Add support for application credentials based authentication. #4968
+* [ENHANCEMENT] Add metric for number of rule groups loaded. #5090
+* [BUGFIX] Avoid duplicate tests for alert unit tests. #4964
+* [BUGFIX] Dont depend on given order when comparing samples in alert unit testing. #5049
```
Comment (brian-brazil) [inline CHANGELOG.md]: Don't

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-rc.0 / 2019-01-18
+
+This release adds support for PromQL Subquries and disk size based retention.
```
Comment (simonpasquier) [inline CHANGELOG.md]: subqueries

Comment (brian-brazil) [inline CHANGELOG.md]: We only include a preamble where there's a significant change, which none of these are. The disk warning should go on the appropriate line

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,26 @@
+## 2.7.0-rc.0 / 2019-01-18
+
+This release adds support for PromQL Subquries and disk size based retention. 
+Please note that the disk size calculations currently only account for space 
+occupied by blocks, and not the WAL (which could be of a significant size).
+
+* [FEATURE] Add Subqueries to PromQL #4831
```
Comment (brian-brazil) [inline CHANGELOG.md]: Subqueries isn't a proper noun, so should be lowercase

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+* [FEATURE] Add support for disk size based retention. #5109 prometheus/tsdb#343
+* [FEATURE] Add CORS origin flag. #5011
+* [ENHANCEMENT] TSDB: Run CI tests on Windows too to catch windows specific errors. prometheus/tsdb#469
```
Comment (simonpasquier) [inline CHANGELOG.md]: not sure we need this and it was already there for v2.6.0

Comment (brian-brazil) [reply CHANGELOG.md]: We don't mention things that aren't user visible, like this

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
+* [ENHANCEMENT] Kubernetes SD: Add service external IP and external name to the discovery metadata. #4940
+* [ENHANCEMENT] Azure SD: Add support for Managed Identity authentication. #4590
+* [ENHANCEMENT] Azure SD: Add tenant and subscription IDs to the discovery metadata. #4969
+* [ENHANCEMENT] OpenStack SD: Add support for application credentials based authentication. #4968
+* [ENHANCEMENT] Add metric for number of rule groups loaded. #5090
+* [BUGFIX] Avoid duplicate tests for alert unit tests. #4964
+* [BUGFIX] Dont depend on given order when comparing samples in alert unit testing. #5049
+* [BUGFIX] Make sure the retention period doesn't overflow. #5112
+* [BUGFIX] Make sure the blocks dont get very large. #5112
+* [BUGFIX] Dont generate blocks with no samples. prometheus/tsdb#374
```
Comment (simonpasquier) [inline CHANGELOG.md]: Don't

File: CHANGELOG.md (hunk 2 above)
Comment (gouthamve) [reply CHANGELOG.md]: Hmm, I think both subqueries and an highly anticipated _experimental_ change should be highlighted. Atleast that it's experimental.

**Comment by gouthamve** [comment]
@brian-brazil @simonpasquier PTAL. I updated the wording and feedback.

File: CHANGELOG.md (hunk 2 above)
Comment (brian-brazil) [reply CHANGELOG.md]: I'd expect only big breaking stuff to be mentioned, I wouldn't want to fatigue users so they miss one of those.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,23 @@
+## 2.7.0-rc.0 / 2019-01-18
+
+This release adds support for PromQL subqueries and disk size based retention. Please note that the disk size calculations currently only account for space occupied by blocks, and not the WAL (which could be of a significant size). You can set both time and disk size retention policies, with time based retention defaulting to 15d when unset. Further, the disk based retention is an experimental feature with no stability guarantees.
```
Comment (brian-brazil) [inline CHANGELOG.md]: Details on specific features either belongs in the docs, or on the relevant line below.

Comment (simonpasquier) [inline CHANGELOG.md]: If we go with a preamble, I wouldn't mention subqueries.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,23 @@
+## 2.7.0-rc.0 / 2019-01-18
+
+This release adds experimental support for disk size based retention. To accomodate that we are deprecating the flag `storage.tsdb.retention` in favour of `storage.tsdb.retention.time`. We print a warning if the flag is in use, but it will function without breaking until Prometheus 3.0.
+
+* [FEATURE] Add subqueries to PromQL #4831
```
Comment (simonpasquier) [inline CHANGELOG.md]: missing dot

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**